lib/utils/webpack.config.js

```diff
@@ -0,0 +1,197 @@
+    } else {
+        result = false
+    }
+
+    return result
+}
+
+module.exports = function(program, directory, stage, webpackPort, routes) {
+    webpackPort = webpackPort || 1500;
+    routes = routes || [];
+    function output() {
+        switch(stage) {
+        case "develop":
+            return {
+                path: directory,
+                filename: 'bundle.js',
+                publicPath: `http://${program.host}:${webpackPort}/`
+            }
+        case "production":
+            return {
+                filename: "bundle.js",
+                path: directory + "/public"
+            }
+        case "static":
+            return {
+                path: directory + "/public",
+                filename: "bundle.js",
+                libraryTarget: 'umd'
+            }
+        }
+    }
+
+    function entry() {
+        switch(stage) {
+        case "develop":
+            return [
+                `${__dirname}/../../node_modules/webpack-dev-server/client?${program.host}:${webpackPort}`,
+                `${__dirname}/../../node_modules/webpack/hot/only-dev-server`,
+                `${__dirname}/web-entry`
+            ]
+        case "production":
+            return [
+                `${__dirname}/web-entry`
+            ]
+        case "static":
+            return [
+                `${__dirname}/static-entry`
+            ]
+        }
+    }
+
+    function plugins() {
+        switch(stage) {
+        case "develop":
+            return [
+                new webpack.HotModuleReplacementPlugin(),
+                new webpack.DefinePlugin({
+                    "process.env": {
+                        NODE_ENV: JSON.stringify(process.env.NODE_ENV ? process.env.NODE_ENV : "development")
+                    },
+                    __PREFIX_LINKS__: program.prefixLinks
+                })
+            ]
+        case "production":
+            return [
+                new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
+                new webpack.DefinePlugin({
+                    "process.env": {
+                        NODE_ENV: JSON.stringify(process.env.NODE_ENV ? process.env.NODE_ENV : "production")
+                    },
+                    __PREFIX_LINKS__: program.prefixLinks
+                }),
+                new webpack.optimize.DedupePlugin(),
+                new webpack.optimize.UglifyJsPlugin()
+            ]
+        case "static":
+            return [
+                new StaticSiteGeneratorPlugin('bundle.js', routes),
+                new webpack.DefinePlugin({
+                    "process.env": {
+                        NODE_ENV: JSON.stringify(process.env.NODE_ENV ? process.env.NODE_ENV : "production")
+                    },
+                    __PREFIX_LINKS__: program.prefixLinks
+                })
+            ]
+        }
+    }
+
+    function resolve() {
+        return {
+            extensions: ['', '.js', '.jsx', '.cjsx', '.coffee', '.json', '.less', '.toml', '.yaml'],
```

Comment:
### the object way

Using an object in `merge` will add the `extensions` values to the array.

```
config.merge({
    resolve: {
      extensions: ['ahhh', '.bbb']
    }
})
```

### the function way

Using a function yields more control, but when I tried it it threw up on [glob-pages](https://github.com/gatsbyjs/gatsby/blob/700594d76c359570a922efa23d95318d00010fc9/lib/utils/glob-pages.coffee#L23) later in the build process. That said, it *did* work and in the following example, it added `ahhh` and `.bbb` then removed `ahhh`.

```
    config.merge({
      resolve: {
        extensions: ['ahhh', '.bbb']
      }
    })
    config.merge(function(cfg) {
      cfg.resolve.extensions = _.without(cfg.resolve.extensions, 'ahhh');
      return cfg;
    })
```

Also, I think I've tracked down the issue in webpack configurator for the function-based merging.